### PR TITLE
Make the pylint stricter

### PR DESCRIPTION
**Pull request by Felixoid**
### Changelog category (leave one):
- CI Fix or Improvement (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
...

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [ ]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All Builds
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ]  Exclude: All with aarch64, release, debug
---
- [ ]  Run only fuzzers related jobs (libFuzzer fuzzers, AST fuzzers, etc.)
- [ ]  Exclude: AST fuzzers
---
- [ ]  Do not test
- [ ]  Woolen Wolfdog
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

**Comment by robot-ch-test-poll1** [comment]
*This is an automated comment for commit 49ff0345e383f9526e1685a1a4332b8a47419d97 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/70013/49ff0345e383f9526e1685a1a4332b8a47419d97/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Builds</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/ad2ee0fdf753986754d372a1f0af6177cbe44c0e/builds/report.html">✅ success</a></td></tr>
<tr><td>ClickBench</td><td>Runs [ClickBench](https://github.com/ClickHouse/ClickBench/) with instant-attach table</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/ad2ee0fdf753986754d372a1f0af6177cbe44c0e/clickbench__aarch64_.html">✅ success</a></td></tr>
<tr><td>Compatibility check</td><td>Checks that <b>clickhouse</b> binary runs on distributions with old libc versions. If it fails, ask a maintainer for help</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/ad2ee0fdf753986754d372a1f0af6177cbe44c0e/compatibility_check__aarch64_.html">✅ success</a></td></tr>
<tr><td>Docker keeper image</td><td>The check to build and optionally push the mentioned image to docker hub</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/ad2ee0fdf753986754d372a1f0af6177cbe44c0e/docker_keeper_image.html">✅ success</a></td></tr>
<tr><td>Docker server image</td><td>The check to build and optionally push the mentioned image to docker hub</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/ad2ee0fdf753986754d372a1f0af6177cbe44c0e/docker_server_image.html">✅ success</a></td></tr>
<tr><td>Docs check</td><td>Builds and tests the documentation</td><td>✅ success</td></tr>
<tr><td>Fast test</td><td>Normally this is the first check that is ran for a PR. It builds ClickHouse and runs most of <a href="https://clickhouse.com/docs/en/development/tests#functional-tests">stateless functional tests</a>, omitting some. If it fails, further checks are not started until it is fixed. Look at the report to see which tests fail, then reproduce the failure locally as described <a href="https://clickhouse.com/docs/en/development/tests#functional-test-locally">here</a></td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/ad2ee0fdf753986754d372a1f0af6177cbe44c0e/fast_test.html">✅ success</a></td></tr>
<tr><td>Flaky tests</td><td>Checks if new added or modified tests are flaky by running them repeatedly, in parallel, with more randomization. Functional tests are run 100 times with address sanitizer, and additional randomization of thread scheduling. Integration tests are run up to 10 times. If at least once a new test has failed, or was too long, this check will be red. We don't allow flaky tests, read <a href="https://clickhouse.com/blog/decorating-a-christmas-tree-with-the-help-of-flaky-tests/">the doc</a></td><td>✅ success</td></tr>
<tr><td>Install packages</td><td>Checks that the built packages are installable in a clear environment</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/ad2ee0fdf753986754d372a1f0af6177cbe44c0e/install_packages__aarch64_.html">✅ success</a></td></tr>
<tr><td>Integration tests</td><td>The integration tests report. In parenthesis the package type is given, and in square brackets are the optional part/total tests</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/ad2ee0fdf753986754d372a1f0af6177cbe44c0e/integration_tests__aarch64__[1_6].html">✅ success</a></td></tr>
<tr><td>Performance Comparison</td><td>Measure changes in query performance. The performance test report is described in detail <a href="https://github.com/ClickHouse/ClickHouse/tree/master/docker/test/performance-comparison#how-to-read-the-report">here</a>. In square brackets are the optional part/total tests</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/ad2ee0fdf753986754d372a1f0af6177cbe44c0e/performance_comparison__release__[1_4]/report.html">✅ success</a></td></tr>
<tr><td>Stateful tests</td><td>Runs stateful functional tests for ClickHouse binaries built in various configurations -- release, debug, with sanitizers, etc</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/ad2ee0fdf753986754d372a1f0af6177cbe44c0e/stateful_tests__aarch64_.html">✅ success</a></td></tr>
<tr><td>Stateless tests</td><td>Runs stateless functional tests for ClickHouse binaries built in various configurations -- release, debug, with sanitizers, etc</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/ad2ee0fdf753986754d372a1f0af6177cbe44c0e/stateless_tests__aarch64_.html">✅ success</a></td></tr>
<tr><td>Stress test</td><td>Runs stateless functional tests concurrently from several clients to detect concurrency-related errors</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/ad2ee0fdf753986754d372a1f0af6177cbe44c0e/stress_test__debug_.html">✅ success</a></td></tr>
<tr><td>Unit tests</td><td>Runs the unit tests for different release types</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/ad2ee0fdf753986754d372a1f0af6177cbe44c0e/unit_tests__asan_.html">✅ success</a></td></tr>
<tbody>
</table>
</details>

**Comment by Felixoid** [comment]
The integration tests failed because the rabbit didn't start on time https://pastila.nl/?000290ab/dd0808e05f45c8973fcdc5bd98e7ed47#aGQxVUz/A0a5VxoJBsXvIg== https://s3.amazonaws.com/clickhouse-test-reports/70013/f413f459af050ae74d2ac0b0bf05262e5cb84997/integration_tests__tsan__[4_6].html

File: pyproject.toml [hunk 1]
```diff
@@ -7,42 +7,34 @@ max-branches=50
...
-src_paths = ["src", "tests/ci", "tests/sqllogic"]
-
 [tool.pylint.FORMAT]
 #ignore-long-lines = (# )?<?https?://\S+>?$
 
 [tool.pylint.'MESSAGES CONTROL']
           # pytest.mark.parametrize is not callable (not-callable)
 disable = '''
-          pointless-string-statement,
-          line-too-long,
```
Comment (maxknv) [inline pyproject.toml:21]: this is excessive check for ci scripts

File: pyproject.toml [hunk 2]
```diff
@@ -7,42 +7,34 @@ max-branches=50
...
-          unused-wildcard-import,
-          singleton-comparison,
-          not-callable,
           redefined-outer-name,
           broad-except,
           bare-except,
-          no-else-return,
-          global-statement,
-          f-string-without-interpolation,
-          consider-using-with,
```
Comment (maxknv) [inline pyproject.toml:42]: this is excessive check for ci scripts

File: pyproject.toml [hunk 3]
```diff
@@ -7,42 +7,34 @@ max-branches=50
...
-          use-maxsplit-arg,
```
Comment (maxknv) [inline pyproject.toml:43]: this is excessive check for ci scripts

File: pyproject.toml (hunk 2 above)
Comment (Felixoid) [reply pyproject.toml:42]: This is a must-have practice to keep the resources private in the scope of functions. I found the only single place where it complained, and in the place the check was abused https://github.com/ClickHouse/ClickHouse/pull/70013/files#diff-0397dcb4f24f20fe4b2d7925174332bb6e84f670da123876c12544872e5b1c3fR251

File: pyproject.toml (hunk 3 above)
Comment (Felixoid) [reply pyproject.toml:43]: Again, this one was abused in the place pylint found it https://github.com/ClickHouse/ClickHouse/pull/70013/files#diff-b3f5f3ea620ee8e310f6bdfab21608735e58c4d04402cd5a70fd267047b36a1aR575-R592

Using `maxsplit` is a good practice to save the memory.

File: pyproject.toml (hunk 1 above)
Comment (Felixoid) [reply pyproject.toml:21]: It's opinionated. We have 200 lines limit, and it's not nice to read long lines. Many of places where it complained, became better

[one](https://github.com/ClickHouse/ClickHouse/pull/70013/files#diff-b3f5f3ea620ee8e310f6bdfab21608735e58c4d04402cd5a70fd267047b36a1aR72)
[two](https://github.com/ClickHouse/ClickHouse/pull/70013/files#diff-b3f5f3ea620ee8e310f6bdfab21608735e58c4d04402cd5a70fd267047b36a1aR466-R469)
[three](https://github.com/ClickHouse/ClickHouse/pull/70013/files#diff-4a5684a1ef65d804c900f98ec655040f7e5bfff1b446b2509364fb9419a2342fR159-R167)

In any case, if you disagree, you can enable it locally in the file. But have it globally enabled for the whole repo is the idea I disagree

**Comment by Felixoid** [comment]
Ok, the only one remaining point is `line-too-long`, and we agreed it can be added on the file level if needed

**Pull request merged**